integrations/api_repo_git_commits_test.go

```diff
@@ -30,3 +30,15 @@ func TestAPIReposGitCommits(t *testing.T) {
 	req := NewRequestf(t, "GET", "/api/v1/repos/%s/repo1/git/commits/unknown?token="+token, user.Name)
 	session.MakeRequest(t, req, http.StatusNotFound)
 }
+
+func TestAPIReposGitCommitList(t *testing.T) {
+	prepareTestEnv(t)
+	user := models.AssertExistsAndLoadBean(t, &models.User{ID: 2}).(*models.User)
+	// Login as User2.
+	session := loginUser(t, user.Name)
+	token := getTokenForLoggedInUser(t, session)
+
+	// Test getting non-existent refs
+	req := NewRequestf(t, "GET", "/api/v1/repos/%s/repo1/git/commits/?token="+token, user.Name)
+	session.MakeRequest(t, req, http.StatusOK)
```

Comment:
Also assert the JSON result is better.

Reply:
Okay, I also added a few more test cases (requesting an empty page, requesting a different branch)